c7n/actions/webhook.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+import jmespath
+import requests
+from six.moves.urllib import parse
+
+from c7n import utils
+from .core import EventAction
+
+
+class Webhook(EventAction):
+    """Calls a webhook with optional parameters and body
+       populated from JMESPath queries.
+
+        .. code-block:: yaml
+
+          policies:
+            - name: call-webhook
+              resource: ec2
+              description: |
+                Call webhook with list of resource groups
+              actions:
+               - type: webhook
+                 url: http://foo.com
+                 parameters:
+                    - resource_name: resource.name
+                    - policy_name: policy.name
+    """
+
+    schema = utils.type_schema(
+        'webhook',
+        required=['url'],
+        **{
+            'url': {'type': 'string'},
+            'body': {'type': 'string'},
+            'batch': {'type': 'boolean'},
+            'batch-size': {'type': 'number'},
+            'method': {'type': 'string', 'enum': ['PUT', 'POST', 'GET', 'PATCH', 'DELETE']},
```

Comment:
I would the policy will fail validation if you put a `method` that is not a part of this enum, but a test may be helpful.

Reply:
done